tools/dgeni/common/directive-metadata.ts

```diff
@@ -0,0 +1,74 @@
+import {CategorizedClassDoc} from './dgeni-definitions';
+import {
+  ArrayLiteralExpression,
+  CallExpression,
+  ObjectLiteralExpression,
+  PropertyAssignment,
+  StringLiteral, SyntaxKind
+} from 'typescript';
+
+export type DirectiveMetadata = Map<string, any>;
+
+/**
+ * Determines the component or directive metadata from the specified Dgeni class doc. The resolved
+ * directive metadata will be stored in a Map.
+ *
+ * Currently only string literal assignments and array literal assignments are supported.
```

Comment:
Add "Other value types are not necessary because they are not needed for any user-facing documentation"

Reply:
Done :+1: